File: names.py
#!/usr/bin/env python
#
# Copyright (c) 2011 jesus m. rodriguez
#
# This software is licensed to you under the GNU General Public License,
# version 2 (GPLv2). There is NO WARRANTY for this software, express or
# implied, including the implied warranties of MERCHANTABILITY or FITNESS
# FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
# along with this software; if not, see
# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
#

import sys

lastname = None

if len(sys.argv) < 2:
    print "Usage: names.py [filename]"
    sys.exit(-1)

if len(sys.argv) > 2:
    lastname = sys.argv[2]

file = open(sys.argv[1])

names = []
while 1:
    line = file.readline()
    name = line.rstrip()
    if name != '':
        names.append(name)
    if not line:
        break

names.sort()

for firstname in names:
    for middlename in names:
        if firstname != middlename:
            if not lastname:
                print "%s %s" % (firstname, middlename)
            else:
                print "%s %s %s - %s%s%s" % (
                    firstname, middlename, lastname,
                    firstname[:1], middlename[:1], lastname[:1])

File: names.go
/*
 * Copyright (c) 2018 jesus m. rodriguez
 *
 * This software is licensed to you under the GNU General Public License,
 * version 2 (GPLv2). There is NO WARRANTY for this software, express or
 * implied, including the implied warranties of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
 * along with this software; if not, see
 * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
 */

package main

import (
	"bufio"
	"fmt"
	"io"
	"os"
	"sort"
)

var lastname string

func main() {
	var names []string

	if len(os.Args) < 2 {
		fmt.Println("Usage: names [filename]")
		os.Exit(-1)
	}

	if len(os.Args) > 2 {
		lastname = os.Args[2]
	}

	file, err := os.Open(os.Args[1])
	if err != nil {
		fmt.Println(err)
		os.Exit(-1)
	}

	rdr := bufio.NewReader(file)

	for {
		line, _, err := rdr.ReadLine()
		if err != nil {
			if err != io.EOF {
				fmt.Printf("error reading file: %v\n", err)
			}
			break
		}

		name := string(line)
		if name != "" {
			names = append(names, name)
		}
		if line == nil {
			break
		}
	}

	sort.Strings(names)

	for _, firstname := range names {
		for _, middlename := range names {
			if firstname != middlename {
				if lastname == "" {
					fmt.Printf("%s %s\n", firstname, middlename)
				} else {
					fmt.Printf("%s %s %s - %s%s%s\n",
						firstname, middlename, lastname,
						firstname[:1], middlename[:1], lastname[:1])
				}
			}
		}
	}
}
